includes/meta.php

```diff
@@ -0,0 +1,237 @@
+
+
+/**
+ * Prints Chartbeat tracking code in the footer if a UID and Domain are set in
+ * the customizer.
+ **/
+function add_chartbeat() {
+	$uid = get_theme_mod_or_default( 'chartbeat_uid' );
+	$domain = get_theme_mod_or_default( 'chartbeat_domain' );
+
+	if ( $uid && $domain ) {
+?>
+<script type="text/javascript">
+    var _sf_async_config = _sf_async_config || {};
+    /** CONFIGURATION START **/
+    _sf_async_config.uid = '<?php echo $uid; ?>'
+    _sf_async_config.domain = '<?php echo $domain; ?>';
+    /** CONFIGURATION END **/
+    (function() {
+        function loadChartbeat() {
+            var e = document.createElement('script');
+            e.setAttribute('language', 'javascript');
+            e.setAttribute('type', 'text/javascript');
+            e.setAttribute('src', '//static.chartbeat.com/js/chartbeat.js');
+            document.body.appendChild(e);
+        }
+        var oldonload = window.onload;
+        window.onload = (typeof window.onload != 'function') ?
+            loadChartbeat : function() {
+                oldonload();
+                loadChartbeat();
+            };
+    })();
+</script>
+<?php
+	}
+}
+
+add_action( 'wp_footer', 'add_chartbeat' );
+
+
+/**
+ * Prints the Google Tag Manager data layer snippet in the document head if a
+ * GTM ID is set in the customizer.
+ **/
+function google_tag_manager_dl() {
+	$gtm_id = get_theme_mod_or_default( 'gtm_id' );
+	if ( $gtm_id ) :
+?>
+<script>
+	dataLayer = [];
+</script>
+<?php
+	endif;
+}
+
+add_action( 'wp_head', 'google_tag_manager_dl', 2 );
+
+
+/**
+ * Prints the Google Tag Manager script tag in the document head if a GTM ID is
+ * set in the customizer.
+ **/
+function google_tag_manager() {
+	$gtm_id = get_theme_mod_or_default( 'gtm_id' );
+	if ( $gtm_id ) :
+?>
+<!-- Google Tag Manager -->
+<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+'//www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+})(window,document,'script','dataLayer','<?php echo $gtm_id; ?>');</script>
+<!-- End Google Tag Manager -->
+<?php
+	endif;
+}
+
+add_action( 'wp_head', 'google_tag_manager', 3 );
+
+
+/**
+ * Prints the Google Tag Manager noscript snippet using the GTM ID in Theme Options.
+ **/
+function google_tag_manager_noscript() {
+	$gtm_id = get_theme_mod_or_default( 'gtm_id' );
+	if ( $gtm_id ) :
+?>
+<!-- Google Tag Manager (noscript) -->
+<noscript><iframe src="//www.googletagmanager.com/ns.html?id=<?php echo $gtm_id; ?>"
+height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
+<!-- End Google Tag Manager (noscript) -->
+<?php
+	endif;
+}
+
+add_action( 'after_body_open', 'google_tag_manager_noscript', 0 );
+
+
+/**
+ * Generates a title based on context page is viewed.  Stolen from Thematic
+ **/
+function header_title( $title, $separator ) {
```

Comment:
Are we treating this as a fallback, i.e. Yoast will override these if a title is set?